examples/03_connectivity/plot_confounds_removal.py

```diff
@@ -0,0 +1,167 @@
+                                    current_volume_motion[:-1]))
+confounds = np.hstack((current_volume_motion, previous_volume_motion,
+                       current_volume_motion ** 2,
+                       previous_volume_motion ** 2))
+
+##########################################################################
+# Anatomical CompCor
+# ----------------------------------------------
+# Signals within noisy ROIs, such as WM and CSF, are assumed to reflect noise.
+# The anatomical CompCor approach removes the first principal components
+# of noisy ROIs.
+#
+# First, we define the tissue masks.
+# As we do not have an anatomical image, we are relying on group-level masks.
+# Note that subject-specific masks would give more accurate results.
+
+##########################################################################
+# We give the locations of probabilistic group masks
+icbm152 = datasets.fetch_icbm152_2009()
+mask_images = {'gm': icbm152.gm, 'wm': icbm152.wm, 'csf': icbm152.csf}
+
+##########################################################################
+# and use them to compute binary masks.
+from nilearn import image
+for tissue_name, threshold in zip(['wm', 'gm', 'csf'], ['.9', '.5', '.3']):
+    mask_images[tissue_name] = image.math_img('img > ' + threshold,
+                                              img=mask_images[tissue_name])
+
+##########################################################################
+# We erode WM and CSF masks, to avoid including any signal of neuronal origin.
+from scipy import ndimage
+for tissue_name in ['wm', 'csf']:
+    mask_img = mask_images[tissue_name]
+    eroded_mask_data = mask_img.get_data()
+    for n_erosions in range(2):
+        eroded_mask_data = ndimage.binary_erosion(eroded_mask_data)
+    mask_images[tissue_name] = image.new_img_like(mask_img, eroded_mask_data)
+
+##########################################################################
+# Next, we resample the tissue masks to the functional image
+from nilearn import image, masking
+mean_func_img = image.mean_img(func_filename)
+func_mask_img = masking.compute_epi_mask(mean_func_img)
+affine = mean_func_img.get_affine()
+shape = mean_func_img.get_data().shape
+for tissue_name, mask_img in mask_images.items():
+    mask_images[tissue_name] = image.resample_img(mask_img, target_shape=shape,
+                                                  target_affine=affine,
+                                                  interpolation='nearest')
+
+##########################################################################
+# and intersect them with a brain mask computed from the mean functional image.
+from nilearn import masking
+func_mask_img = masking.compute_epi_mask(mean_func_img, opening=0)
+for tissue_name, mask_img in mask_images.items():
+    mask_images[tissue_name] = image.math_img("img1 * img2", img1=mask_img,
+                                              img2=func_mask_img)
```

Comment:
We have special function to apply mask (`masking.apply_mask` I think). Also some functions, such as `threshold_img` can take mask as parameters.